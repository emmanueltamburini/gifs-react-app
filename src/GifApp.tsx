import { useState } from "react";
import AddCategory from "./Components/AddCategory";

const GifApp = () => {
  const [categories, setCategories] = useState<string[]>(['One Punch', 'Dragon ball']);

  const onAddCategory = () => {
    setCategories(state => ["Demon Slayer", ...state]);
  }

  return (
    <>
        <h1>GifApp</h1>
        <AddCategory />
        <button onClick={onAddCategory}>Add</button>

        <ol>
            {categories.map(element => <li key={element}>{element}</li>)}
        </ol>
    </>
  )
}

export default GifApp