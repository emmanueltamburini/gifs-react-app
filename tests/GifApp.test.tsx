import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import GifApp from '../src/GifApp';

describe('Test in GifApp', () => {
    test('should be equal to snapshot', () => {
        const {container} = render(<GifApp />);
        expect(container).toMatchSnapshot();
    });

    test('should add a new category', () => {
        const category = 'category1';
        render(<GifApp />);
        fireEvent.change(screen.getByPlaceholderText('Search gif'), { target: { value: category } });
        fireEvent.submit(screen.getByLabelText('form'));
        expect(screen.getByText(category));
    });

    test('should not add new category with empty search', () => {
        const category = '     ';
        render(<GifApp />);
        fireEvent.change(screen.getByPlaceholderText('Search gif'), { target: { value: category } });
        fireEvent.submit(screen.getByLabelText('form'));
        expect(screen.queryByText(category)).toBeNull();
        expect(screen.queryByText('Loading...')).toBeNull();
        expect(screen.queryAllByRole('img').length).toBe(0);
    });

    test('should not add new category if it is repeated', () => {
        const category = 'category2';
        const repeatCategory = 3;
        render(<GifApp />);
        for (let i = 0; i < repeatCategory; i++) {
            fireEvent.change(screen.getByPlaceholderText('Search gif'), { target: { value: category } });
            fireEvent.submit(screen.getByLabelText('form'));
        }
        expect(screen.getAllByText(category).length).toBe(1);
    });
})