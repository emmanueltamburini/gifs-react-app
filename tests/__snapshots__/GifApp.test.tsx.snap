// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifApp should be equal to snapshot 1`] = `
<div>
  <h1>
    GifApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gif"
      type="text"
      value=""
    />
  </form>
</div>
`;
